common/basecomponent/base.go

```diff
@@ -184,16 +256,46 @@ func (b *BaseDendrite) CreateKeyDB() keydb.Database {
 	if err != nil {
 		logrus.WithError(err).Panicf("failed to connect to keys db")
 	}
+	if b.LibP2P != nil {
```

Comment:
Please don't do this. Instead, do what I have done for `dendritejs` - pull out this code to the `main.go` package and just duplicate the few lines above which does:
```
db, err := keydb.NewDatabase(
		string(b.Cfg.Database.ServerKey),
		string(b.Cfg.Database.ServerKey),
		b.Cfg.Matrix.ServerName,
		b.Cfg.Matrix.ServerName,
		b.Cfg.Matrix.PrivateKey.Public().(ed25519.PublicKey),
		b.Cfg.Matrix.PrivateKey.Public().(ed25519.PublicKey),
		b.Cfg.Matrix.KeyID,
		b.Cfg.Matrix.KeyID,
	)
	)
```

At the moment we're just coupling p2p to the base component for no reason.